Sources/Tracking/DeviceInfo.swift

```diff
@@ -1,15 +1,56 @@
 import Foundation
 #if canImport(UIKit)
 import UIKit
+import UserNotifications
 
-internal enum DeviceInfo {
-    /// Device's model on which SDK is running eg. iPhone12,3
-    static let deviceInfo: String = UIDevice.deviceModelCode
-    /// Operating system and version of OS of the Device
-    static let osInfo: String = "\(UIDevice.current.systemName) \(UIDevice.current.systemVersion)"
-    /// Name of customer's application using the SDK
-    static let customerAppName: String = Bundle.main.infoDictionary?["CFBundleName"] as? String ?? ""
-    /// Version of the customer's application using the SDK
-    static let customerAppVersion: String = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+/// To get basic detail about the device SDK is working on
+/// such as operating system, customer app name, bundle id etc
+public enum DeviceInfo {
+    case deviceInfo
+    case osInfo
+    case customerAppName
+    case customerAppVersion
+    case customerBundleId
+    case sdkVersion
+    case deviceLocale
+    public var value: String {
+        switch self {
+        case .deviceInfo:
+            return UIDevice.deviceModelCode
+        case .osInfo:
+            return "\(UIDevice.current.systemName) \(UIDevice.current.systemVersion)"
+        case .customerAppName:
+            return Bundle.main.infoDictionary?["CFBundleName"] as? String ?? ""
+        case .customerAppVersion:
+            return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+        case .customerBundleId :
+            return Bundle.main.bundleIdentifier ?? ""
+        case .sdkVersion:
+            return SdkVersion.version
+        case .deviceLocale:
+            return Locale.current.identifier
+        }
+    }
+}
+
+/// Provides additional async details about device the SDK is working on
+public class DeviceDetail {
```

Comment:
I find it confusing to have a class `DeviceInfo` and a class `DeviceDetail` where it seems to me that the only reason that they are not all in 1 class is because you can't make `pushSubscribed()` synchronous and therefore, cannot put it inside of the `DeviceInfo` enum. 

Is there a different way that we can implement capturing device attributes so avoid having 2 separate classes when they each seem to have the same purpose?

Reply:
`DeviceInfo` is an enum and `DeviceDetail` is a class. But i think i should club both of them together as even I find it absurd to have two entities performing almost same function. Doing it right away. Changes can be found in latest commit.

Reply:
✔️